daft/dataframe/dataframe.py

```diff
@@ -65,6 +67,17 @@
 ManyColumnsInputType = Union[ColumnInputType, Iterable[ColumnInputType]]
 
 
+def BroadcastMetrics(func):
+    """Calls `self._explain_broadcast()` prior to executing the rest of the function."""
+
+    def wrapper(self, *args, **kwargs):
+        result = func(self, *args, **kwargs)
+        self._explain_broadcast()
```

Comment:
Does this mean that the dashboard only receives the broadcast message after the method finishes execution? Seems wrong.

Another way you can potentially solve this is with a context manager in each dataframe execution method. Something like this:

```python
def write_parquet():
    df = ... # build plan
    with ctx.dashboard.record_execution(df):
        ...
        return df
```